backend/app/api/endpoints/internal/rag.py

```
@@ -901,3 +921,120 @@ async def read_document(
+    generated_sql: str = Field(default="", description="Generated SQL query")
+    explanation: str = Field(default="", description="SQL explanation")
+    confidence: float = Field(default=0.0, description="Generation confidence")
+    results: StructuredQueryResult = Field(default_factory=StructuredQueryResult)
+    sources: list[dict] = Field(default_factory=list)
+    error: Optional[str] = Field(default=None)
+
+
+@router.post("/structured-query", response_model=StructuredQueryResponse)
+async def structured_query(
+    request: StructuredQueryRequest,
+    db: Session = Depends(get_db),
```

Ruff (0.15.0)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)